### PR TITLE
design-process: mandate a design for all feature branches.

### PR DESCRIPTION
**Pull request by paulidale**
If it's large enough to need a feature branch, it should be designed.

### CODE REVIEW

File: policies/design-process.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ Requirement for Design Documents
 
 A design document is required for any proposed enhancement which adds
 significant new APIs or non-trivially evolves or modifies existing APIs.
+A design document is also required all feature branches.
```
Comment (mattcaswell) [inline policies/design-process.md:44]: "for all"?

Comment (paulidale) [reply policies/design-process.md]: Ugh.  I read that sentence literally five or six times looking for mistakes 😢

**Review by mattcaswell — APPROVED** [review]

**Comment by t-j-h** [comment]
I think the update is entirely redundant. 
I also think we can have feature branches for purposes other than adding new APIs ... but that's a different topic.

I don't see that this adds value to change and especially when this technical policy is **not** being followed currently (in that we have many new APIs coming in through PRs that do not have design documents.

**Comment by paulidale** [comment]
This has absolutely nothing to do with APIs.  A new API does **not** currently mandate a design document and review process.  That is not being changed here.  As an aside: we might be pushing the _trivial_ API definition too far and if so, we should fix the wording of the policy to clarify.

What is being made explicit is that work deemed sufficiently large/complicated to require a feature branch is also large/complex enough to require a design document.  No more no less.  It doesn't matter if APIs are involved, it's purely about the size/complexity of the change.

**Comment by paulidale** [comment]
Calling vote:
```
Topic: Accept the design process policy update as of 839bc6b
Proposed by: Pauli
Issue link: https://github.com/openssl/technical-policies/pull/105
Public: yes
Opened: 2024-11-12
```

**Comment by paulidale** [comment]
Pauli: [+1]

**Comment by kroeckx** [comment]
Voting +1

**Comment by t8m** [comment]
Vote: [0]

**Comment by beldmit** [comment]
Vote: [0]

**Comment by levitte** [comment]
vote: [+0]

**Comment by mattcaswell** [comment]
+1

**Comment by slontis** [comment]
+1

**Comment by paulidale** [comment]
The vote passes.

```Topic: Accept the design process policy update as of 839bc6b
Proposed by: Pauli
Issue link: https://github.com/openssl/technical-policies/pull/105
Public: yes
Opened: 2024-11-12
Closed: 2024-11-19
Accepted:  yes  (for: 4, against: 0, abstained: 3, not voted: 3)

  Dmitry     [ 0]
  Matt       [+1]
  Tim        [  ]
  Hugo       [  ]
  Richard    [+0]
  Shane      [+1]
  Tomas      [ 0]
  Kurt       [+1]
  Pauli      [+1]
  Nicola     [  ]
```

**Comment by t-j-h** [comment]
Note that the vote has closed 

Vote: [-1]

**Comment by paulidale** [comment]
It's closed and you are outside the two week voting window.

**Comment by t8m** [comment]
Do we really observe the two week window for the post-closing votes??

**Pull request closed without merging**